test/gilded-rose.spec.ts

```diff
@@ -4,6 +4,275 @@ describe('Gilded Rose', () => {
+    expect(items[0].quality).toBe(7);
+  })
+
+  it('given Backstage passes is in inventory and sellIn 5 days, when 1 days passes, then quality increase by 3', () => {
+    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 5, 5)]);
+
+    let items;
+    for (let i = 0; i < 1; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(8);
+  })
+
+  it('given Backstage passes is in inventory and sellIn 0 days, when 1 days passes, then quality becomes 0', () => {
+    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 0, 5)]);
+
+    let items;
+    for (let i = 0; i < 1; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(0);
+  })
+
+  it('given Conjured Mana Cake is in inventory, when 10 days passes, then quality decreases by 20', () => {
+    const gildedRose = new GildedRose([new Item('Conjured Mana Cake', 15, 30)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(10);
+  })
+
+  it('given Conjured Mana Cake is in inventory and sellIn is negative, when 5 days passes, then quality decreases by 20', () => {
+    const gildedRose = new GildedRose([new Item('Conjured Mana Cake', -5, 30)]);
+
+    let items;
+    for (let i = 0; i < 5; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(10);
+  })
+
+  it('given Conjured Mana Cake is in inventory and sellIn is 5, when 10 days passes, then quality decreases by 30', () => {
+    const gildedRose = new GildedRose([new Item('Conjured Mana Cake', 5, 40)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(10);
+  })
+
+  it('given Conjured Mana Cake is in inventory and quality is 5, when 10 days passes, then quality is 0', () => {
+    const gildedRose = new GildedRose([new Item('Conjured Mana Cake', 15, 5)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(0);
+  })
+
+
+  // kevin test cases
+
+  it('Normal items degrade by 1 unit per day before expiring', () => {
+    // Given
+    const gildedRose = new GildedRose([new Item('Normal Item', 1, 10)]);
+
+    // When
+    const items = gildedRose.updateQuality();
+
+    // Then
+    expect(items[0].quality).toBe(9);
+  });
+
+  it('Normal items degrade by 2 unit per day after expiring', () => {
+    // Given
+    const gildedRose = new GildedRose([new Item('Normal Item', 0, 10)]);
+
+    // When
+    const items = gildedRose.updateQuality();
+
+    // Then
+    expect(items[0].quality).toBe(8);
+  });
+
+  it('Normal items degrade by 1 unit before expiring, then expires and degrades by 2 units', () => {
+    // Given
+    const gildedRose = new GildedRose([new Item('Normal Item', 1, 10)]);
+
+    // When
+    const items = new GildedRose(gildedRose.updateQuality()).updateQuality();
```

Comment:
I think I'd prefer it if this line is split into two - first line updates the quality of the original gilded rose, second one does the update again.